### PR TITLE
[REVIEW] Add BlazingSQL version of query 04

### PR DESCRIPTION
**Pull request by Christian8491**
This PR adds the version of "query" 04 using `BlazingSQL`.

Contains: `reduction_function()` and `abandonedShoppingCarts()` functions from `tpcx_bb_query_04.py`

### CODE REVIEW

**Comment by Christian8491** [comment]
It shows **Correctness Assertion True** at scale factor **1K**, using **dgx201**.

```
(tpcx) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q04$ python tpcx_bb_query_04_sql.py --data_dir=/raid/tpcx-bb/sf1000/parquet_2gb/ --verify_results --verify_dir=/mnt/weka/nicholasb/tpcx-bb/validation/sf1000/ --cluster_host=10.150.162.27
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: Count not make directory
WARNING:root:WARNING: Count not make directory
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
Scalar Result Query
Correctness Assertion True
Please install gspread and oauth2client               to use Google Sheets automation
```

**Comment by Christian8491** [comment]
Updated and tested again at sf **1K** on the **dgx201**. It shows **Correctness Assertion True** .

```
(tpcx) christianc@rl-dgx2-d17-u05-rapids-dgx201:~/tpcx-bb/tpcx_bb/queries/q04$ python tpcx_bb_query_04_sql.py --config_file=/home/christianc/yaml_dir/validation_1k_config.yaml
Using default arguments
Connected to tcp://10.150.162.27:8786
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.protobuf.UnsafeUtil (file:/home/christianc/miniconda3/envs/tpcx/lib/blazingsql-algebra.jar) to field java.nio.Buffer.address
WARNING: Please consider reporting this to the maintainers of com.google.protobuf.UnsafeUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
BlazingContext ready
Scalar Result Query
Correctness Assertion True
Please install gspread and oauth2client               to use Google Sheets automation
```

from `validation_1k_config.yaml`, some useful values:

```
data_dir: /raid/tpcx-bb/sf1000/parquet_2gb/
verify_results: True
verify_dir: /mnt/weka/nicholasb/tpcx-bb/validation/sf1000/
```

File: tpcx_bb/queries/q04/tpcx_bb_query_04_sql.py [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import sys
+
+from xbb_tools.cluster_startup import attach_to_cluster
+from xbb_tools.sessionization import get_sessions
+
+import cudf
```
Comment (beckernick) [inline tpcx_bb/queries/q04/tpcx_bb_query_04_sql.py:23]: Can we push this down to local scope inside a function?

Comment (Christian8491) [reply tpcx_bb/queries/q04/tpcx_bb_query_04_sql.py]: Updated.

**Pull request merged**